frontend/src/views/instructor/assignments/index.tsx

```diff
@@ -1,18 +1,99 @@
 import React from "react";
-import { ConnectedOfferTable } from "../../admin/offertable";
+import { ConnectedTAsTable } from "./tas-table";
 import { ConnectedExportAssignmentsAction } from "../../admin/assignments/import-export";
 import { ActionsList, ActionHeader } from "../../../components/action-buttons";
 import { ContentArea } from "../../../components/layout";
+import { useSelector } from "react-redux";
+import {
+    activeSessionSelector,
+    assignmentsSelector,
+} from "../../../api/actions";
+import { ddahsSelector, upsertDdah } from "../../../api/actions/ddahs";
+import { Ddah } from "../../../api/defs/types";
+import { MissingActiveSessionWarning } from "../../../components/sessions";
+import { useThunkDispatch } from "../../../libs/thunk-dispatch";
+import { activePositionSelector } from "../store/actions";
+import { DdahPreviewModal } from "../ddahs/ddah-editor";
 
 export function InstructorAssignmentsView() {
+    const activeSession = useSelector(activeSessionSelector);
+    const dispatch = useThunkDispatch();
+    const activePosition = useSelector(activePositionSelector);
+    const assignments = useSelector(assignmentsSelector);
+
+    const [previewVisible, setPreviewVisible] = React.useState<Boolean>(false);
+    const [newDialogVisible, setNewDialogVisible] = React.useState<Boolean>(
+        false
+    );
+    const [previewDdah, setPreviewDdah] = React.useState<Omit<
+        Ddah,
+        "id"
+    > | null>(null);
+
     return (
         <div className="page-body">
             <ActionsList>
                 <ActionHeader>Import/Export</ActionHeader>
                 <ConnectedExportAssignmentsAction />
             </ActionsList>
             <ContentArea>
-                <ConnectedOfferTable editable={false} />
+                {activeSession ? null : (
+                    <MissingActiveSessionWarning extraText="To view or modify DDAHs, you must select a session." />
+                )}
+                <ConnectedTAsTable
+                    position_id={activePosition?.id || -1}
+                    onViewDDAH={(ddah) => {
+                        if (ddah) {
+                            setPreviewDdah(ddah);
+                            setPreviewVisible(true);
+                        }
+                    }}
+                    onCreateDDAH={(assignment_id: number) => {
+                        const assignment = assignments.find(
+                            (a) => a.id === assignment_id
+                        );
+                        if (!assignment) {
```

Comment:
This is not yet cleaned up - some of this logic may be abstracted away in separate components, other logic can be simplified/changed.